ThirdPartyAdapters/vungle/vungle/src/main/java/com/google/ads/mediation/vungle/VungleBaseAdapter.java

```diff
@@ -0,0 +1,146 @@
+package com.google.ads.mediation.vungle;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.text.TextUtils;
+import android.util.Log;
+import androidx.annotation.NonNull;
+import com.google.ads.mediation.vungle.VungleInitializer.VungleInitializationListener;
+import com.google.android.gms.ads.mediation.InitializationCompleteCallback;
+import com.google.android.gms.ads.mediation.MediationAdLoadCallback;
+import com.google.android.gms.ads.mediation.MediationConfiguration;
+import com.google.android.gms.ads.mediation.MediationInterstitialAd;
+import com.google.android.gms.ads.mediation.MediationInterstitialAdCallback;
+import com.google.android.gms.ads.mediation.MediationInterstitialAdConfiguration;
+import com.google.android.gms.ads.mediation.VersionInfo;
+import com.google.android.gms.ads.mediation.rtb.RtbAdapter;
+import com.google.android.gms.ads.mediation.rtb.RtbSignalData;
+import com.google.android.gms.ads.mediation.rtb.SignalCallbacks;
+import com.vungle.mediation.BuildConfig;
+import com.vungle.warren.Vungle;
+import java.util.HashSet;
+import java.util.List;
+
+public abstract class VungleBaseAdapter extends RtbAdapter {
+
+  private static final String TAG = VungleBaseAdapter.class.getSimpleName();
+  protected static final String KEY_APP_ID = "appid";
+
+  @Override
+  public void collectSignals(@NonNull RtbSignalData rtbSignalData,
+      @NonNull SignalCallbacks signalCallbacks) {
+    String token = Vungle.getAvailableBidTokens(rtbSignalData.getContext());
```

Comment:
@HuFangshuai whats the testing setup for this, I never was able to get this call to fire

Reply:
yes, me too. Seems we need some setup on AdMob dashboard or adUnitIds.